src/configurationProvider.ts

```diff
@@ -290,4 +356,29 @@ function convertLogLevel(commonLogLevel: string) {
 interface IMainClassOption {
     readonly projectName?: string;
     readonly mainClass: string;
+    readonly filePath?: string;
+}
+
+interface IMainClassQuickPick extends vscode.QuickPickItem {
+    item: IMainClassOption;
+}
+
+class RecentlyUsedHistory {
+    private cache: { [key: string]: number } = {};
```

Comment:
Use the Map for this this purpose?

Reply:
This data structure has no relationship with Recently Used. Will refine this interface to an appropriate name.